src/freenet/clients/fcp/ClientRequest.java

```diff
@@ -27,6 +27,9 @@
  */
 public abstract class ClientRequest implements Serializable {
 
+    /**
+     * ATTENTION: When incrementing this, please skip version 2. Version 2 had already temporarily
+     * been used by a development branch. */
```

Comment:
OK, I've removed the excessive documentation at f1f71fefbd78fb25bb1ee556ae5528e1fe16d217

> (Is there even a situation where we're willing to change the UID? I thought that broke backwards compatibility?)

The purpose of UIDs is to break compatibility, yes. This is intended, and scheduled for regular use: If we ever need to change the class architecture in a way which would be incompatible with pre-existing serialized data, then we would change the UID. This would ensure that the deserializer would refuse to load the existing serialized data. The node would then falll back to the backup of the download queue which it stores in a different format. 
The reason why the UID changing was not used in this case is that the backup code has the disadvantage of only being implemented for the download-queue, not for the upload queue.
I would by the way suggest that we soon implement backup for uploads as well to prevent future huge discussions of whether to change or not change the UID. We've already had this discussion now even though the code was only released as testing version. In the future, it could very well be necessary that we break compatibility to the released official code.